feugy/melodie
# feat: uses jwt to secure accesses

### What's in there?

In the previous implementation, successfully connecting a single WebSocket unlocks access to the media files and data, regardless of the origin.

This PR introduces JWT tokens as a way to secure access. Clients need a TOTP to establish WS connection, and receive a token and settings in return. 

Token must be used when invoking core functions through WS, accessing cover and artworks, and downloading the track data of course.

Token have a 30 minutes expiry, and the core automatically sends a new token before expiry. They can also be used when reconnecting WS, which saves UI from receiving and storing new TOTP every 30 seconds, and provide a more reliable user experience.

Receiving settings upon WS connection makes UI init simpler and faster.

Are included here:

- [x] feat(core): uses JWT to authenticate incoming requests (media, WS messages)
- [x] feat(ui): stores JWT upon connection
- [x] refactor(ui): initialized settings upon connection
- [x] refactor(ui): uses enhanceUrl() instead of global variable
- [x] refactor(ui): do not store Totp
- [x] chore(ui): fixes atelier mocks and styles
- [x] chore(desktop): displays log file in logs

### How to test?

Start the app, enables broadcast, and load Mélodie on a browser.
Try different scenario of disconnection, and see when TOTP is required, and when JWT is automaticaly reused.
